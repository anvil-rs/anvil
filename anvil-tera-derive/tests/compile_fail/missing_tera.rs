// This file tests the error when a template is missing the required tera attribute
// Expected error: "Missing 'tera' attribute within #[template(...)]"

use anvil_tera_derive::Template;
use serde::Serialize;

#[derive(Serialize, Template)]
#[template(path = "test.txt")]  // Missing tera attribute
struct MissingTeraTemplate {
    value: String,
}

fn main() {
    // This will never run, we're only testing the compile-time error
}
