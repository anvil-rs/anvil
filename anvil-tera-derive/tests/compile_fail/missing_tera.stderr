error: Missing 'tera' attribute within #[template(...)]. Example: #[template(..., tera = MY_TERA_INSTANCE)]
  --> $DIR/missing_tera.rs:8:1
   |
8  | #[derive(Serialize, Template)]
   | ^
